### DISCUSSION TITLE
Solution for replacing ~/.cargo/env ?

### DISCUSSION THREAD

**Discussion by wiiznokes** [fish-shell/fish-shell#10577] [General]
Somehow, i can't use rustup anymore. Before today, it worked. I think this is because rustup write `~/.bashrc` with `. "$HOME/.cargo/env"`

```
$ cat ~/.cargo/env 
#!/bin/sh
# rustup shell setup
# affix colons on either side of $PATH to simplify matching
case ":${PATH}:" in
    *:"$HOME/.cargo/bin":*)
        ;;
    *)
        # Prepending path in case a system-installed rustc needs to be overridden
        export PATH="$HOME/.cargo/bin:$PATH"
        ;;
esac
```
So fish doesn't have access to it. It's really weird because i swear i could use rustup in fish before.

I there a proper fix for it ? Could rustup use another method to support fish oob ?

**Comment by zanchey** [comment]
rustup also writes `.config/fish/conf.d/rustup.fish` - is that file there?

**Comment by wiiznokes** [comment]
No, there is nothing in `~/.config/fish/conf.d/rustup.fish`. It must have been removed somehow... I wonder if there is a command to know what happened. So i written this based on [this file](https://github.com/rust-lang/rustup/blob/master/src/cli/self_update/env.fish)

```fish
# rustup shell setup
if not contains "$HOME/.cargo/bin" $PATH
    # Prepending path in case a system-installed rustc needs to be overridden
    set -x PATH "$HOME/.cargo/bin" $PATH
end
```

**Reply by Rhylan2333** [reply]
Thanks! It works. I wrote the same content based on [this file](https://github.com/rust-lang/rustup/blob/master/src/cli/self_update/env.fish) and saved it as ~/.config/fish/conf.d/env.fish

**Comment by thomasa88** [comment]
I did not have fish installed when I installed Rust, so I was also missing rustup.fish. I reran the install script/line from https://www.rust-lang.org/tools/install and got the file:

`curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh`

**Comment by tibitTW** [comment]
The replacement should be `source "$HOME/.cargo/env.fish"` while using fish shell
